v2/internal/reconcilers/arm/arm_client.go

```diff
@@ -51,7 +47,7 @@ type Connection struct {
 func newConnection(client *armClient) *Connection {
```

Comment:
Instead of this being a bare function that accepts a client, consider making it a method on `armClient`.

```suggestion
func (client *armClient) newConnection() *Connection {
```

Reply:
Done